plugins/woocommerce-blocks/assets/js/base/utils/legacy-events.ts

```diff
@@ -59,6 +59,13 @@ export const triggerAddedToCartEvent = ( {
 	} );
 };
 
+export const triggerProductListRenderedEvent = (): void => {
+	dispatchEvent( 'wc-blocks_product_list_rendered', {
+		bubbles: true,
+		cancelable: true,
+	} );
+};
+
```

Comment:
Just out of curiosity, why are we adding this new event in the `legacy-events.ts` file? Is this new event a legacy event, or are we just adding it to this file because it's related to events? 🙂

Reply:
>Is this new event a legacy event, or are we just adding it to this file because it's related to events?

The latter. I don't think creating a new place for it would help in maintenance so wanted to keep them in one place.

However, in the meantime, I kicked off a refactor on that area: https://github.com/woocommerce/woocommerce/pull/50199 (branched out from this branch) to extract a separate `@woocommerce/base-events` package. I'm discussing it with @gigitux elsewhere. But the reactor is not blocking delivery of the event itself hence I'm keeping the separated.

I hope that makes sense

Reply:
Totally makes sense! Thanks you for kicking off the refactor. I really like how you are creating ENUM for event names 💜